### PR TITLE
fix(contract) add subscribe to event methods to contract interface

### PR DESCRIPTION
**Pull request by tkmct**

### CODE REVIEW

File: src/contract/interfaces/IDepositContract.ts [hunk 1]
```diff
@@ -19,4 +19,21 @@ export interface IDepositContract {
    * @param depositedRangeId The id of range. We can know depositedRangeId from deposited event and finalizeExited event.
    */
   finalizeExit(exit: Property, depositedRangeId: Integer): Promise<void>
+
+  /**
+   * subscribe to deposit events
+   */
+  subscribeDeposit(
```
Comment (syuhei176) [inline src/contract/interfaces/IDepositContract.ts]: It seems good! but there isn't deposit event now. When someone deposit asset, "CheckpointFinalized" event will be fired.

Comment (tkmct) [reply src/contract/interfaces/IDepositContract.ts]: so checkpoint finalized event and deposit event is homogeneous event?

Comment (tkmct) [reply src/contract/interfaces/IDepositContract.ts]: can we handle them in the same way?

Comment (syuhei176) [reply src/contract/interfaces/IDepositContract.ts]: I'm not confident now but current contract implementation is following this spec.
https://docs.plasma.group/projects/spec/en/latest/src/02-contracts/deposit-contract.html#deposit

File: src/contract/interfaces/IDepositContract.ts [hunk 2]
```diff
@@ -19,4 +19,19 @@ export interface IDepositContract {
...
    */
   finalizeExit(exit: Property, depositedRangeId: Integer): Promise<void>
+
+  /**
+   * subscribe to checkpoint finalized event
+   */
+  subscribeCheckpointFinalized(
+    handler: (
+      checkpointId: Bytes,
+      checkpoint: [[number, number], Property]
```
Comment (syuhei176) [inline src/contract/interfaces/IDepositContract.ts]: ah this may be bigint.

Comment (tkmct) [reply src/contract/interfaces/IDepositContract.ts]: ok

**Pull request merged**